### PR TITLE
Add artifactType

### LINKED ISSUE #5: Move artifact-spec go types here
Move these here: https://github.com/oci-playground/artifact-spec/tree/main/specs-go/v1

See https://github.com/oci-playground/artifact-spec/issues/6

### PR DESCRIPTION
**Pull request by sajayantony**
Fixes https://github.com/oci-playground/image-spec/issues/5

### CODE REVIEW

File: specs-go/v1/artifact.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	MediaType string `json:"mediaType"`
+
+	// ArtifactType is the IANA type of this object.
+	ArtifactType string `json:"artifactType"`
+
+	// Blobs is a collection of blobs referenced by this manifest.
+	Blobs []Descriptor `json:"blobs,omitempty"`
+
+	// Refers is an optional link to any existing manifest within the repository.
+	Refers Descriptor `json:"refers,omitempty"`
```
Comment (jdolitsky) [inline specs-go/v1/artifact.go]: Should this be a pointer as fixed in https://github.com/oci-playground/artifact-spec/pull/4 ?

**Review by jdolitsky — APPROVED** [review]

**Comment by sudo-bmitch** [comment]
We also need `artifactType` in the descriptor for the API responses. https://github.com/oci-playground/image-spec/blob/pr/descriptor.md

File: specs-go/v1/artifact.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2016 The Linux Foundation
```
Comment (jdolitsky) [inline specs-go/v1/artifact.go]: 2022 :)

**Comment by sajayantony** [comment]
> We also need `artifactType` in the descriptor for the API responses. https://github.com/oci-playground/image-spec/blob/pr/descriptor.md

@sudo-bmitch I'll make another PR just to append the `artifactType` to the descriptor. Is there any concern naming this type Artifact instead of ArtifactManifest. Let me know and I can rename the file and the type. Just looked more consistent with types in Image Spec.

**Review by jdolitsky — APPROVED** [review]

File: specs-go/v1/artifact.go (hunk 2 above)
Comment (sajayantony) [reply specs-go/v1/artifact.go]: :) Fixed. I was not sure if it was something to do with the license header.

**Review by sudo-bmitch — APPROVED** [review]

**Review by sudo-bmitch — CHANGES_REQUESTED** [review]
The filename is fine. Change the target branch from main to pr please.

**Comment by sajayantony** [comment]
@sudo-bmitch - updated to push these to the PR. I'll follow up with the descriptor change. I assume all this will be open to feedback and we will rework these accordingly.

**Review by sudo-bmitch — APPROVED** [review]

**Pull request merged**